src/diffusers/schedulers/scheduling_heun_discrete.py

```diff
@@ -244,6 +281,7 @@ def step(
         timestep: Union[float, torch.FloatTensor],
         sample: Union[torch.FloatTensor, np.ndarray],
         return_dict: bool = True,
+        step_index: Optional[int] = None,
```

Comment:
```suggestion
```

Ok for now, but this needs to be refactored before merging. We should not add a `step_index` function argument to the `step` function. The `step_index` should be retrieved from the `self.index_for_timestep` function.

Reply:
@patrickvonplaten 

Actually, not sure we can infer `step_index` from `timestep`

Here is the code from the original repo that creates `timesteps` based on `sigma`, note that the `dtype` is `long` - that means we will have the same `t` for multiple steps, therefore can't retrieve the exact index. I will leave it here for now. Let me know how we should handle this 


```
def denoise(self, x_t, sigmas, clip_denoised=True, model_kwargs=None):
        t = th.tensor(
            [self.sigma_to_t(sigma) for sigma in sigmas.cpu().numpy()],
            dtype=th.long,
            device=sigmas.device,
        )
       ...
```
https://github.com/openai/shap-e/blob/main/shap_e/diffusion/k_diffusion.py#L100

Reply:
Just because dtype = `torch.long` doesn't mean we have the same t -> note that we just need to convert sigmas back to timesteps here